src/server/game/Entities/Player/Player.cpp

```diff
@@ -13909,13 +13909,19 @@ void Player::OnGossipSelect(WorldObject* source, uint32 gossipListId, uint32 men
             SendRespecWipeConfirm(guid, sWorld->getBoolConfig(CONFIG_NO_RESET_TALENT_COST) ? 0 : GetNextResetTalentsCost(), SPEC_RESET_PET_TALENTS);
             break;
         case GossipOptionNpc::DisableXPGain:
+        {
             PlayerTalkClass->SendCloseGossip();
+            CastSpell(source->ToPlayer(), 206662, true); // Experience Eliminated
```

Comment:
no magic numbers :)

Reply:
Source is NOT player, this will always be nullptr